src/core/emotionInterface/emotionInterfaceModule.cpp

```diff
@@ -500,19 +537,29 @@ bool EmotionInterfaceModule::setMouth(const std::string cmd)
 bool EmotionInterfaceModule::setEyelids(const std::string cmd)
 {
     char cmdbuffer[] = {0,0,0,0};
-    int i; 
-    i = getIndex(cmd);
+    const auto i = getIndex(cmd);
     if(i < 0)
         return false;
 
+    auto res{ true };
+
     if( _emotion_table[i].eli[0] == '*' || _emotion_table[i].eli[1] == '*') 
         return true;  //leave it in the same state
+    if (_iPos)
+    {
+        std::string percentage_str{ _emotion_table[i].eli[0] + '.' + _emotion_table[i].eli[1] };
+        auto target_pos = std::stod(percentage_str) * (_max-_min);
+        res = _iPos->positionMove(_joint_eylids, target_pos);
+    }
+    else
+    {
+        cmdbuffer[0] = 'S';
+        cmdbuffer[1] = _emotion_table[i].eli[0];
+        cmdbuffer[2] = _emotion_table[i].eli[1];
+        res = writePort(cmdbuffer);
+    }
```

Comment:
If I'm not mistaken, this `else` should check if the robot doesn't have the RFE (in case we'll raise only a warning for RFE robots w/ motors off).

Reply:
In this `else` fall:

- rfe robots that have issues in the face motors(broken or turn off)
- old robots that commands the eyelids through serial.

Unfortunately there is not a clear way to understand inside the module if the robot has or not the rfe

Reply:
Ok, if we don't have an easy way to check whether a robot is RFE, then we stick to this 👍